### ISSUE TITLE
Enforce word alignment in glyf (loca) table

### ISSUE THREAD

**Issue by vadim-160102** [silnrsi/font-ttf#12]
Please see:
https://github.com/ssimms/pdfapi2/issues/37
https://github.com/aliftype/amiri/issues/225

Unfortunately, existing "unaligned" fonts can't be brushed aside as invalid, and "should" in the TTF spec isn't polite variant of "must", of course. In practice, only font subsetting for PDF with Font::TTF is ever expected to be affected (both PDF::API2 and PDF::Reuse suffer), any other scenarios are very unlikely.

Easiest fix would be to add
`$self->{' DAT'} .= "\000" if length($self->{' DAT'}) % 2;`
immediately after https://github.com/silnrsi/font-ttf/blob/c0bd356e70fdb37504dca0bb7f2d9f7ce4b9a8d0/lib/Font/TTF/Glyph.pm#L420

I think it will make 2 lines https://github.com/silnrsi/font-ttf/blob/c0bd356e70fdb37504dca0bb7f2d9f7ce4b9a8d0/lib/Font/TTF/Glyph.pm#L655-L656 redundant, though I don't understand the reason for double-word alignment there, and #657 seems to have been redundant already, but maybe I'm missing something.

**Comment by mhosken** [comment]
Yes (your line insertion for line 420 that word aligns each glyph) is what is needed.

No it doesn't make the 4 byte boundary test for the whole table redundant.

Thank you. Code updated.
